lib/ansiblelint/formatters/__init__.py

```diff
@@ -87,6 +87,33 @@ def format(self, match, colored=False):
                                     match.message)
 
 
+class AnnotationsFormatter(BaseFormatter):
+    # https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-a-warning-message
+    """Emits errors in format that is recognized by Github Annotation parsing.
+
+    ::warning file={name},line={line},col={col}::{message}
+
+    Supported levels: debug, warning, error
+    """
+
+    def format(self, match, colored=False):
```

Comment:
Let's explicitly state the invocation contract:
```suggestion
    def format(self, match, colored=False):
        """Prepare a match instance for reporting as a GitHub Actions annotation."""
        if colored:
            raise ValueError('The colored mode is not supported.')

```